### PR TITLE
Add autoValueGson

### PR DESCRIPTION
**Pull request by SalmaSamy**
AutoValueGson allows deserializing objects created from Auto_value classes

### CODE REVIEW

File: MODULE.bazel [hunk 1]
```diff
@@ -29,9 +29,13 @@ maven.install(
...
+    "com.squareup:javapoet:1.12.0",
     "com.google.j2objc:j2objc-annotations:1.3",
     "com.github.stephenc.jcip:jcip-annotations:1.0-1",
     "org.checkerframework:checker-qual:3.12.0",
+    "com.ryanharter.auto.value:auto-value-gson-extension:1.3.1",
+    "com.ryanharter.auto.value:auto-value-gson-runtime:1.3.1",
+    "com.ryanharter.auto.value:auto-value-gson-factory:1.3.1",
   ],
   repositories = [
     "https://dl.google.com/android/maven2",
```
Comment (Wyverald) [inline MODULE.bazel:41]: shouldn't we uncomment the `lock_file` line?

Comment (SalmaSamy) [reply MODULE.bazel:41]: Done

Comment (Wyverald) [reply MODULE.bazel:41]: remove the TODO too

**Review by Wyverald — APPROVED** [review]

File: third_party/java/javapoet/BUILD [hunk 2]
```diff
@@ -13,8 +13,7 @@ filegroup(
     srcs = glob(["**"]),
 )
 
-distrib_java_import(
+java_library(
     name = "javapoet",
-    enable_distributions = ["debian"],
-    jars = ["javapoet-1.8.0.jar"],
```
Comment (Wyverald) [inline third_party/java/javapoet/BUILD:19]: we should probably also remove this jar

**Comment by sgowroji** [comment]
Hi @Wyverald , I could see above PR has third-party changes. Should i merge the third-party changes first and do the remaining through CL or vice versa ?

**Comment by Wyverald** [comment]
For this one, you'll probably need to merge the non-third-party changes first and then do the third-party ones. Thank you!

**Comment by sgowroji** [comment]
Hi @SalmaSamy, Could you please squash all the commits into one.  Thanks!

**Comment by SalmaSamy** [comment]
Hi @sgowroji, squashing the commits is done!

**Comment by Wyverald** [comment]
See https://github.com/bazelbuild/bazel/pull/17379 instead

**Pull request closed without merging**